test/e2e/next-script/index.test.ts

```diff
@@ -31,9 +31,67 @@ describe('beforeInteractive', () => {
             )
           }
         `,
-        'pages/index.js': `
+        'pages/index.js': `        
+          export default function Home() {
+            return (
+              <>
+                <p>Home page</p>
+              </>
+            )
+          }
+        `,
+      },
+      dependencies: {
+        react: '17.0.2',
+        'react-dom': '17.0.2',
+      },
+    })
+  })
+  afterAll(() => next.destroy())
+
+  it('Script is injected server-side', async () => {
+    let browser: BrowserInterface
+
+    try {
+      browser = await webdriver(next.url, '/')
+
+      const script = await browser.eval(
+        `document.querySelector('script[data-nscript="beforeInteractive"]')`
+      )
+      expect(script).not.toBeNull()
+    } finally {
+      if (browser) await browser.close()
+    }
+  })
+})
+
+describe('beforeInteractive in document body', () => {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    next = await createNext({
+      files: {
+        'pages/_document.js': `
+          import { Html, Head, Main, NextScript } from 'next/document'
           import Script from 'next/script'
-        
+          
+          export default function Document() {
+            return (
+              <Html>
+                <Head />
+                <body>
+                  <Main />
+                  <NextScript />
+                  <Script
```

Comment:
Is it expected that a `next/script` `beforeInteractive` tag defined in the body would be moved the `head` automatically? This may confuse users that do want it to be loaded in the body unless we warn for this case or keep it in place. 

<img width="1176" alt="Screen Shot 2022-06-24 at 11 21 31 AM" src="https://user-images.githubusercontent.com/22380829/175576664-ce7d9707-dace-47a5-979f-71cdd65d9bf9.png">

Reply:
Yeah the idea behind `beforeInteractive` is that it should be used sparingly to load a script before any other script on the page (including Next.js chunks). That was always the behavior before we begin enforcing users to place it in `_document.js`, regardless of which component/page it was placed in. I worry that we'll change its intrinsic behavior if we begin to inject scripts in `body` if the user starts placing things there.

However, I understand that it can be confusing for users to place a script in `body` but not see it outputted that way in the DOM. Should I add a console and/or ESLint warning for this? (e.g. "`beforeInteractive scripts will always be inserted into the document head. If you would like to inject a script into the body instead, do this instead...`")

CC @janicklas-ralph for his thoughts too.

Reply:
Added a note 🙏